pkg/replacer/image/image.go

```diff
@@ -202,6 +229,54 @@ func GetImageDigestFromRef(ctx context.Context, imageRef, platform string, cache
 	}, nil
 }
 
-func shouldExclude(ref string) bool {
-	return ref == "scratch"
+func shouldSkipImageRef(cfg *config.Config, ref string) bool {
+	// Parse the image reference
+	nameRef, err := name.ParseReference(ref)
+	if err != nil {
+		// we wouldn't know how to resolve this reference, so let's skip
+		return true
+	}
+
+	imageName := getImageNameFromRef(nameRef)
+	if slices.Contains(cfg.Images.ImageFilter.ExcludeImages, imageName) {
+		return true
+	}
+
+	tag := nameRef.Identifier()
+	return slices.Contains(cfg.Images.ImageFilter.ExcludeTags, tag)
+}
+
+// TODO(jakub): this is a bit of a hack, but I didn't find a better way to get just the name
+func getImageNameFromRef(nameRef name.Reference) string {
```

Comment:
feel free to ignore this comment: What if we add unit tests specifically for this? That way, if someone refactors and comes up with a better way of doing it, it can be easy to spot problems or validate if the solution worked.

Reply:
good suggestion, I'll do it in a follow-up